src/bsp/platform/stm32fx_nucleo_wifi/xi_bsp_io_net_stm32fx_nucleo_wifi.c

```diff
@@ -179,37 +193,30 @@ xi_bsp_io_net_state_t xi_bsp_io_net_select( xi_bsp_socket_events_t* socket_event
 
     size_t socket_id = 0;
 
-    if ( xi_net_state.wifi_state == xi_wifi_state_connected )
+    for ( socket_id = 0; socket_id < socket_events_array_size; ++socket_id )
     {
-        for ( socket_id = 0; socket_id < socket_events_array_size; ++socket_id )
-        {
-            xi_bsp_socket_events_t* socket_events = &socket_events_array[socket_id];
-
-            if ( 1 == socket_events->in_socket_want_read )
-            {
-                if ( NULL != xi_net_state.head )
-                {
-                    socket_events->out_socket_can_read = 1;
-                }
-            }
+        xi_bsp_socket_events_t* socket_events = &socket_events_array[socket_id];
 
-            if ( 1 == socket_events->in_socket_want_connect )
+        if ( 1 == socket_events->in_socket_want_read )
+        {
+            if ( NULL != xi_net_state.head )
             {
-                socket_events->out_socket_connect_finished = 1;
+                socket_events->out_socket_can_read = 1;
             }
+        }
 
-            if ( 1 == socket_events->in_socket_want_write )
-            {
-                socket_events->out_socket_can_write = 1;
-            }
+        if ( 1 == socket_events->in_socket_want_connect )
+        {
+            socket_events->out_socket_connect_finished = 1;
         }
 
-        return XI_BSP_IO_NET_STATE_OK;
-    }
-    else
-    {
-        return XI_BSP_IO_NET_STATE_ERROR;
+        if ( 1 == socket_events->in_socket_want_write )
+        {
+            socket_events->out_socket_can_write = 1;
+        }
     }
+
+    return XI_BSP_IO_NET_STATE_OK;
```

Comment:
just asking and no change request for this here and now: 
So in theory we could remove the HAL_Delay from the wifi_state_idle: state and put this delay conditionally here into the select function. And call this delay only if there is nothing to do: read, write, connect.
Is this true?